frontend/src/metabase/query_builder/components/DimensionList.jsx

```diff
@@ -185,17 +186,21 @@ export default class DimensionList extends Component {
     } = this.props;
     const dimension = useOriginalDimension
       ? item.dimension
-      : item.dimension.defaultDimension() || item.dimension;
+      : // XXX: 7. Important note that dimension with binning options is `dimension.defaultDimension()`
+        item.dimension.defaultDimension() || item.dimension;
     const shouldExcludeBinning =
-      !enableSubDimensions &&
-      !useOriginalDimension &&
-      dimension instanceof FieldDimension &&
-      dimension.binningStrategy();
-
-    if (
-      shouldExcludeBinning ||
-      (preventNumberSubDimensions && dimension.field().isSummable())
-    ) {
+      (!enableSubDimensions &&
+        !useOriginalDimension &&
+        dimension instanceof FieldDimension &&
+        dimension.binningStrategy()) ||
+      (preventNumberSubDimensions && dimension.field().isSummable()) ||
+      dimension.field().isFK();
```

Comment:
This uses a similar check from [#17783](https://github.com/metabase/metabase/pull/17783/files#diff-a747d1635f99a8170f6f7e9bfd95d5a2375e80241fc1c518131ee152805d1406R114)

Basically, this condition should never happen, but this is a rare case of a weird state.